### DISCUSSION TITLE
Fix for "call to virDomainCreateWithFlags failed: Network not found: no network with matching name 'netlab_1'" on Fedora

### DISCUSSION THREAD

**Discussion by Kircheneer** [ipspace/netlab#53] [Ideas]
I recently ran into [this](https://github.com/vagrant-libvirt/vagrant-libvirt/issues/626) issue in vagrant-libvirt to do with [this](https://fedoraproject.org/wiki/Changes/Vagrant_2.2_with_QEMU_Session) change to Vagrant using `qemu:///session` instead of ` qemu:///system` under Fedora. For some reason this causes the Vagrantfile generated by `netlab create` to fail with this error:

```$ vagrant up
Bringing machine 'r1' up with 'libvirt' provider...
Bringing machine 'r2' up with 'libvirt' provider...
==> r1: Checking if box 'CumulusCommunity/cumulus-vx' version '4.4.0' is up to date...
==> r1: Starting domain.
==> r1: An error occurred. The error will be shown after all tasks complete.
==> r2: Checking if box 'CumulusCommunity/cumulus-vx' version '4.4.0' is up to date...
==> r2: Starting domain.
==> r2: An error occurred. The error will be shown after all tasks complete.
An error occurred while executing multiple actions in parallel.
Any errors that occurred are shown below.

An error occurred while executing the action on the 'r1'
machine. Please handle this error then try again:

There was an error talking to Libvirt. The error message is shown
below:

Call to virDomainCreateWithFlags failed: Network not found: no network with matching name 'netlab_1'

An error occurred while executing the action on the 'r2'
machine. Please handle this error then try again:

There was an error talking to Libvirt. The error message is shown
below:

Call to virDomainCreateWithFlags failed: Network not found: no network with matching name 'netlab_2'
```

The fix is to add `domain.qemu_use_session = false` to the hosts in the Vagrantfile.

Now here's the question: I'm not fit enough with QEMU and Vagrant to decide whose "fault" this is and whether a fix for this should be included with this. Any opinions?

**Comment by ipspace** [comment]
Reading the links you provided, I'm guessing that running Vagrant totally unprivileged doesn't give you access to libvirt networks (`virsh net-list` would return empty output) and so it's impossible to create custom networks, which is a must-have functionality for us.

We could add the setting to `Vagrantfile.j2`. Could you try that out? It seems to have no negative impact in my Ubuntu installation.

**Comment by ipspace** [comment]
Implemented in c8bfd3b8af2857141c9f4501e634944a1d5f91cd
